src/Aquarius.ONE.ClientSDK/Operations/Sample/SampleApi.cs

```diff
@@ -116,19 +120,102 @@ public async Task<bool> CreateTestGroupAsync(TestAnalyteGroup testGroup)
             try
             {
                 var respContent = await _restHelper.PostRestProtobufAsync(testGroup, endpoint).ConfigureAwait(_continueOnCapturedContext);
-                if (respContent.ResponseMessage.IsSuccessStatusCode)
+                Event(null,
+                    respContent.ResponseMessage.IsSuccessStatusCode
+                        ? CreateLoggerArgs(EnumEventLevel.Trace, "CreateTestGroupAsync Success", respContent.ResponseMessage.StatusCode, watch.ElapsedMilliseconds)
+                        : CreateLoggerArgs(EnumEventLevel.Warn, "CreateTestGroupAsync Failed", respContent.ResponseMessage.StatusCode, watch.ElapsedMilliseconds));
+
+                if (!respContent.ResponseMessage.IsSuccessStatusCode)
+                    return ErrorResponse(respContent, false);
+
+                return true;
+
+            }
+            catch (Exception e)
+            {
+                Event(e, CreateLoggerArgs(EnumEventLevel.Error, $"CreateTestGroupAsync Failed - {e.Message}"));
+                throw e;

```

Comment:
Based on the teams conversation with @JonathanRobertson-AQI  this should also be a call to the ErrorResponse method providing the caught exception and a default return value.

Reply:
It might be good to have the ErrorResponse method return a custom exception so we do not need to catch things twice.